tests/expectations/tests/libclang-5/generate-c-inline.rs

```diff
@@ -0,0 +1,20 @@
+/* automatically generated by rust-bindgen */
+
+#![allow(
+    dead_code,
+    non_snake_case,
+    non_camel_case_types,
+    non_upper_case_globals
+)]
+
+c! { # include "generate-c-inline.h" }
```

Comment:
Where do these macros come from?

Reply:
[rust-c](https://github.com/lemonrock/rust-c) has a simple `c! {}` [implementation](https://github.com/lemonrock/rust-c/blob/master/src/mac.rs) for C, and [rust-cpp](https://github.com/mystor/rust-cpp) also provides `cpp! {}`[macro for C++](https://docs.rs/cpp/0.5.1/cpp/macro.cpp.html) 

if you think the way is right, I could add a special implementation to `bindgen`, it need add some code to `build.rs`, generate some `.c`/`.cpp` file on building.

Reply:
I think it may be ok to depend on external crates, but then it should be documented and obviously not on-by-default.

Reply:
After read the `rust_c` and `rust-cpp` crates, I believe we only need [a very simple implementation](https://github.com/rust-lang/rust-bindgen/pull/1460/commits/fd1ca6135c2dd9e3128195a932dcc8ac7255b29f#diff-567e947a8cc5ad306d3eff99dfb66cc9R3665) like

```rust
macro_rules! c {
    () => {};

    (#include $filename:tt $($rest:tt)*) => {
        c!{ $($rest)* }
    };

    ({ $body:tt } $($rest:tt)*) => {
        c!{ $($rest)* }
    };
}
```
and a `bindgen-build` crate be added to support [extract and build C/C++ code](https://github.com/rust-lang/rust-bindgen/pull/1460/commits/fd1ca6135c2dd9e3128195a932dcc8ac7255b29f#diff-2f7f1b1e5b6dbd8d0f7067081a62e0feR14) in the build script.

You can check [a real example](https://raw.githubusercontent.com/flier/rust-dpdk/master/rte-sys/src/raw.rs) that includes hundreds of static inline functions with [some build script](https://github.com/flier/rust-dpdk/blob/master/rte-sys/build.rs#L81), like
```rust
    bindgen_build::build(binding_file, "rte", |build| {
        build
            .flag("-march=native")
            .include("src")
            .include(rte_sdk_dir.join("include"));

        for (name, value) in &cpu_features {
            build.define(name, value.as_ref().map(|s| s.as_str()));
        }
    }).unwrap();
```

What's your opinion?

Reply:
besides, with the C/C++ wrapper, we could reexport the thread local variable so solve #1187

Reply:
This approach works, but there's a lot of unanswered questions related to mangling and such...

I'm sorry for having missed this, btw :(

Maybe a better approach would be to instead add a `ParseCallback` when we see a thread-local variable, in order to let the caller to do whatever they want with it, since this seems like a pretty opinionated solution to the problem.

That would allow us not to depend on `c!` / `cpp!` in any way.